### ISSUE TITLE
Schema::combine should return correct schema

### ISSUE THREAD

**Issue by humb1t** [graphgate/graphgate#3]

**Comment by humb1t** [comment]
<img width="1685" alt="image" src="https://github.com/graphgate/graphgate/assets/5635032/c4b6c5cf-cb5a-4964-b52a-6d30c91cd708">
Need more tests on complex subgraphs and their relations.

**Comment by humb1t** [comment]
Right now, combine results depends on the order of subgraphs' arguments passed to the method:

```
    let schema = ComposedSchema::combine([
        ("collectibles".to_string(), collectibles_service_document),
        ("collections".to_string(), collections_service_document),
    ])
    .unwrap();
```
will differ from 
```
    let schema = ComposedSchema::combine([
        ("collectibles".to_string(), collectibles_service_document),
        ("collections".to_string(), collections_service_document),
    ])
    .unwrap();
```

It results in inconsistent behaviour of Graphgate - it may work differently based on which service returned the SDL first.

**Comment by humb1t** [comment]
Fixed in #6
